tests/scripts/connection-error-test.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+#
+# Simulate a 2 node cluster where nodeid2 is unreachable (black hole address).
+# Timed out commands, due to the connect error, will trigger a slot map update.
+#
+# Usage: $0 /path/to/clusterclient-binary
+
+clientprog=${1:-./clusterclient_async_sequence}
+testname=connection-error-test
+
+# Sync processes waiting for CONT signals.
+perl -we 'use sigtrap "handler", sub{exit}, "CONT"; sleep 1; die "timeout"' &
+syncpid1=$!;
+
+# Start simulated redis node #1
+timeout 5s ./simulated-redis.pl -p 7401 -d --sigcont $syncpid1 <<'EOF' &
+EXPECT CONNECT
+EXPECT ["CLUSTER", "SLOTS"]
+SEND [[0, 6000, ["127.0.0.1", 7401, "nodeid1"]],[6001, 16383, ["192.168.254.254", 9999, "nodeid2"]]]
+EXPECT CLOSE
+
+EXPECT CONNECT
+EXPECT ["SET", "bar", "initial"]
+SEND +OK
+
+# Max retry triggers a fetch of config
+EXPECT CONNECT
+EXPECT ["PING"]
+SEND +PONG
+EXPECT ["config", "get", "cluster-node-timeout"]
+SEND ["cluster-node-timeout", "0"]
+# Close this connection since it's unused from now on
+CLOSE
+
+# Topology changed, nodeid2 is now gone
```

Comment:
We can't have two connections to the same fake node in parallel? It would eliminate the possibility that it's the TCP close from this node that triggers CLUSTER SLOTS.

Reply:
The close on line 33 is to close the hiredis-cluster's blocking connect on line 27.
If this connection is not closed the EXPECT on line 41 is not receiving the SET command, since it will read from wrong socket.
To mitigate your worries we can move the close on line 33 to after the CLUSTER SLOTS, but before the EXPECT for the SET command, to line 40.

This testcase will look a bit better after the PR with async slotmap updates, which reuses connections.